File: src/ChainMetrics.UI/app/types/order_product.py
from pydantic import BaseModel, Field


class OrderProduct(BaseModel):
    order_id: str = Field(..., alias="orderId")
    product_id: str = Field(..., alias="productId")
    seller_id: str = Field(..., alias="sellerId")
    price: float = Field(..., alias="price")
    shipping_charges: float = Field(..., alias="shippingCharges")

File: src/ChainMetrics.Domain/Entities/OrderProduct.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ChainMetrics.Domain.Entities;

[Table("df_OrderItems")]
public class OrderProduct
{
    [Column("order_id")]
    public required string OrderId { get; set; }
    public Order? Order { get; set; }

    [Column("product_id")]
    public required string ProductId { get; set; }
    public Product? Product { get; set; }

    [Column("seller_id")]
    public required string SellerId { get; set; }

    [Column("price")]
    public required decimal Price { get; set; }

    [Column("shipping_charges")]
    public required decimal ShippingCharges { get; set; }
}
